libbeat/cmd/keystore.go

```diff
@@ -130,10 +130,15 @@ func createKeystore(settings instance.Settings, force bool) error {
 		return err
 	}
 
+	wKeystore, ok := store.(keystore.WritableKeystore)
+	if !ok {
+		return fmt.Errorf("the configured keystore is not writable")
+	}
```

Comment:
Let's reduce 'duplication' and copy and paste here (I noticed some error message is even wrong :) ) by having a helper function doing the cast:

```
var errNotWritable = errors.New("the configured keystore is not writable")

func asWritableKeystore(store keystore.KeyStore) (keystore.Writable, error) {
  w, ok := store.(keystore.Writable)
  if !ok {
    return nil, errNotWritable
  }
  return w, nil
}
```

maybe we can reduce some stuttering as well? :)